crypto/pem/pem_lib.c

```diff
@@ -654,14 +657,15 @@ int PEM_write_bio(BIO *bp, const char *name, const char *header,
         (BIO_write(bp, name, nlen) != nlen) ||
         (BIO_write(bp, "-----\n", 6) != 6))
         goto err;
-    OPENSSL_clear_free(buf, PEM_BUFSIZE * 8);
-    EVP_ENCODE_CTX_free(ctx);
-    return i + outl;
+    retval = i + outl;
+    /* Fallthrough */
+
  err:
-    OPENSSL_clear_free(buf, PEM_BUFSIZE * 8);
+    if (retval == 0)
+        PEMerr(PEM_F_PEM_WRITE_BIO, reason);
     EVP_ENCODE_CTX_free(ctx);
-    PEMerr(PEM_F_PEM_WRITE_BIO, reason);
-    return 0;
+    OPENSSL_free(buf);
```

Comment:
Why do we no longer need to clear the buffer ?
It could be a private key or something like that, right?

Reply:
We do need to clear it. But the author of this contribution cannot be traced to agree to the licence change. Therefore it has to go. Someone else not familiar with this code will have to put it back.